File: LeetCode/724-Find-Pivot-Index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
        let sum = 0;
        let leftsum = 0;
        let n = nums.length;
 
        for (let i = 0; i < n; i++){
            sum += nums[i];
        }
 
        for (let j = 0; j < n; j++) {
        
            if ( sum - leftsum - nums[j] == leftsum){
                return j;
            }
            leftsum += nums[j];
        }

        return -1;
};

File: LeetCode/724-Find-Pivot-Index.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        int sum = 0;
        int leftsum = 0;
        int n = nums.Length;
 
        for (int i = 0; i < n; ++i)
            sum += nums[i];
 
        for (int i = 0; i < n; ++i) {
 
            sum -= nums[i];
 
            if (leftsum == sum)
                return i;
 
            leftsum += nums[i];
        }
 
        return -1;
    }
}